### DISCUSSION TITLE
Refreshing Dock with Update/Upgrade

### DISCUSSION THREAD

**Discussion by amangalampalli** [Homebrew/discussions#2888] [Everyday usage]
When using `homebrew-autoupdate`, when casks that are pinned to the Dock are updated, it results in the icons randomly disappearing from the Dock. Jetbrains Toolbox does something unique by refreshing the Dock to re-insert the items that have been updated. The same could be employed to `homebrew-autoupdate` to try to refresh said dock and reinsert the same items into the Dock. However, there could be a multitude of challenges with this as I am not sure if the specific order of the applications pinned in the Dock is retained somewhere. Perhaps, by reading this file, `homebrew-autoupdate` could possibly upgrade the applications and keep the icons in the Dock in the same order. Just food for thought, not sure the underlying complexity about something like this.

**Comment by carlocab** [comment]
Related: https://github.com/Homebrew/homebrew-cask/issues/102721
